ifrs17/Utils/EqualityComparers.ipynb

```diff
@@ -91,6 +92,35 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "code",
+            "source": [
+                "class YieldCurveComparer: IEqualityComparer<YieldCurve>",
+                "\n{",
+                "\n    private YieldCurveComparer(){}",
+                "\n",
+                "\n\tpublic bool Equals(YieldCurve x, YieldCurve y) => ",
+                "\n            x.Year == y.Year && x.Month == y.Month && x.Scenario == y.Scenario && x.Currency == y.Currency && x.Id == y.Id && x.Values.SequenceEqual(y.Values); // TODO: enable Precision",
```

Comment:
at the top of this notebook create a SequenceEqual static method extension for double array which accepts another double array plus a double precision. Call this method from all equality comparers in this notebook with the Precision constant.

Reply:
Why do you need to also compare the Values?
Is this only used in the test, or elsewhere?

Reply:
We use the YieldCurveComparer to check if the newly imported Yield Curves are identical to the old ones. This setup saves computational time in case of multiple imports

Reply:
If just for comparison, use Fluent Assertions, as they have a comparer out of the box. See reportVariableTest.

Reply:
Issue created: https://github.com/Systemorph/IFRS17CalculationEngine/issues/146